raft/log.go

```diff
@@ -991,12 +986,12 @@ func (l *Log) applier(done chan chan struct{}) {
 	}
 }
 
-// apply a log initialization command by parsing and setting the configuration.
-func (l *Log) applyInitialize(e *LogEntry) error {
+// mustApplyInitialize a log initialization command by parsing and setting the configuration.
+func (l *Log) mustApplyInitialize(e *LogEntry) {
 	// Parse the configuration from the log entry.
 	config := &Config{}
 	if err := NewConfigDecoder(bytes.NewReader(e.Data)).Decode(config); err != nil {
-		return fmt.Errorf("initialize: %s", err)
+		panic("decode: " + err.Error())
```

Comment:
Yes, if a node can't read the data in the `applier()` loop then there's no way to recover from and no way to return an error (since it's in a separate goroutine). There's some kind of corruption if an error occurs so we just need to bail.